app/routes/errors.py

```diff
@@ -41,9 +44,15 @@ def _render_error_page(status_code, template=None):
     )
 
 
-@errors_blueprint.app_errorhandler(401)
-@errors_blueprint.app_errorhandler(CSRFError)
 @errors_blueprint.app_errorhandler(NoTokenException)
+@errors_blueprint.app_errorhandler(CSRFError)
+@errors_blueprint.app_errorhandler(401)
+def no_cookie(error=None):
+    log_error(error, 401)
+    return _render_error_page(401, "no-cookie")
+
+
+@errors_blueprint.app_errorhandler(NoQuestionnaireStateException)
```

Comment:
This should be one 401 handler with a check for the session cookie, as a cookie can exist and a 401 get raised.